modules/mag-deck/mag-deck-arduino/mag-deck-arduino.ino

```diff
@@ -305,15 +308,22 @@ void activate_bootloader(){
 }
 
 void setup() {
+  memory.read_serial(device_serial);
+  memory.read_model(device_model);
+
+  int model_version = device_model.substring(MODEL_VER_TEMPLATE_LEN).toInt();
+  if (model_version >= 20) {  // Update steps_per_mm for GEN2 or later
+    steps_per_mm = 100;
+    step_delay_microseconds = 100000 / (steps_per_mm * 10);
+  }
+
   setup_pins();
   setup_digipot();
   set_current(CURRENT_HIGH);
   set_speed(SPEED_LOW);
```

Comment:
I think we'll need to adjust the speeds (SPEED_HIGH, SPEED_LOW, SPEED_PROBE) too. The current speed values will make the motor move much faster. The speed settings should be half of GEN1 to keep the actual speeds same in both models.